WordPress/Classes/ViewRelated/Post/PostEditor+Publish.swift

```diff
@@ -197,7 +186,13 @@ extension PostEditor where Self: UIViewController {
         ActionDispatcher.dispatch(NoticeAction.clearWithTag(uploadFailureNoticeTag))
         stopEditing()
 
-        if post.canSave() {
+        /// If a post is marked to be auto uploaded and can be saved, it means that the changes
+        /// had been already confirmed by the user. In this case, we just close the editor.
+        /// Otherwise, we'll show an Action Sheet with options.
+        if post.shouldAttemptAutoUpload && post.canSave() {
```

Comment:
Like my comment below, I'm concerned with the usage of `shouldAttemptAutoUpload` here. However, I could not find a better alternative aside from tracking the “Update” button press or tracking if there were recent changes since the last save. The `Post.hasUnsavedChanges()` also doesn't work for our purpose. 😞 

So, yeah, we can keep this for now. Thank you for adding some comments.

Reply:
Yes, I completely agree with you. :(

I tried some other approaches but they were more complex or error-prone.